service/history/transferQueueTaskExecutorBase.go

```diff
@@ -273,6 +280,12 @@ func (t *transferQueueTaskExecutorBase) deleteExecution(
 		}
 	}
 
+	if checkForPendingCloseExecutionTask {
```

Comment:
IMO, it is cleaner to move this logic into processDeleteExecutionTask and keep deleteExecution() as it is.

Reply:
Definitely, but I wanted it to be there so that I don't have to load mutable state twice.